RecoEgamma/EgammaElectronProducers/plugins/ElectronSeedConverter.cc

```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+#include <string>
+
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/EgammaReco/interface/EleSeedHostCollection.h"
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeed.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeedFwd.h"
+
+
+class ElectronSeedConverter : public edm::global::EDProducer<> {
+public:
+  explicit ElectronSeedConverter(const edm::ParameterSet&);
+
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const final;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+    const edm::EDGetTokenT<TrajectorySeedCollection> initialSeedsToken_;
+    const edm::EDGetTokenT<reco::EleSeedHostCollection> matchedEleSeedSoAToken_; 
+    const edm::EDPutTokenT<reco::ElectronSeedCollection> putToken_;
+	edm::EDGetTokenT<std::vector<reco::SuperClusterRef>> superClustersTokens_;
+};
+
+
+void ElectronSeedConverter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("initialSeeds", {"hltElePixelSeedsCombined"});
+    desc.add<edm::InputTag>("eleSeedsSoA", edm::InputTag{"ElectronNHitSeedAlpakaProducer"});
+  	desc.add<edm::InputTag>("superClusters", {"hltEgammaSuperClustersToPixelMatch"});
+    descriptions.add("electronNHitSeedConverter", desc);
+}
+
+ElectronSeedConverter::ElectronSeedConverter(const edm::ParameterSet& pset)
+    :initialSeedsToken_(consumes(pset.getParameter<edm::InputTag>("initialSeeds"))),
+    matchedEleSeedSoAToken_(consumes<reco::EleSeedHostCollection>(pset.getParameter<edm::InputTag>("eleSeedsSoA"))),    
+    putToken_{produces<reco::ElectronSeedCollection>()} {
+	    superClustersTokens_ = consumes(pset.getParameter<edm::InputTag>("superClusters"));
+    }
```

Comment:
I'd suggest consistency, e.g.
```suggestion
    matchedEleSeedSoAToken_(consumes(pset.getParameter<edm::InputTag>("eleSeedsSoA"))),    
    superClustersTokens_(consumes(pset.getParameter<edm::InputTag>("superClusters"))),
    putToken_(produces()) {}
```